src/rez/tests/test_shells.py

```diff
@@ -183,16 +184,20 @@ def test_rcfile(self):
     @per_available_shell()
     @install_dependent()
     def test_rez_env_output(self):
+        target_shell = config.default_shell  # overridden by test util
+
         def _test(txt):
             # Assumes that the shell has an echo command, build-in or alias
             binpath = os.path.join(system.rez_bin_path, "rez-env")
-            args = [binpath, "--", "echo", txt]
+            args = [binpath, "--shell", target_shell, "--", "echo", txt]
 
             process = subprocess.Popen(
                 args, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
                 stderr=subprocess.PIPE, universal_newlines=True
             )
             sh_out = process.communicate()
+            if sh_out[1]:
+                raise Exception("Command failed:\n%s" % sh_out[1])
```

Comment:
I didn't use `process.returncode` is because, it seems that powershell always return `0` even the command actually failed.

Reply:
Isn't this a case of powershell -File vs. -Command?
Only the latter passes the error code.
https://stackoverflow.com/questions/50200325/returning-an-exit-code-from-a-powershell-script

Or maybe `$ErrorActionPreference = "Stop"`:
https://stackoverflow.com/questions/9948517/how-to-stop-a-powershell-script-on-the-first-error

The checks also need to happen form `$LastExitCode` I believe:
https://stackoverflow.com/questions/10666035/difference-between-and-lastexitcode-in-powershell

What an over-engineered little shell :)